### ISSUE TITLE
Frequent timeouts during deployments

### ISSUE THREAD

**Issue by oakh1** [F5Networks/f5-appsvcs-extension#407]
### Environment
 * Application Services Version: 3.24.0
 * BIG-IP Version: 14.1 (running on viprion chassis)

### Summary
Deployment times out frequently due to poor rest api performance(?)

### Steps To Reproduce
1. Submit any declaration (small, big, doesn't do any difference)

2. Observe the following error response:
```json
{
  "code": 500,
  "message": "connection to http://admin:XXXXXX@localhost:8100/mgmt/tm/asm/policies timed out"
}
```
Or...
```json
{
  "code": 422,
  "message": "declaration failed",
  "response": "connection to http://admin:XXXXXX@localhost:8100/mgmt/tm/task/cli/script run cli script timed out",
  "host": "localhost",
  "tenant": "xxx"
}
```
Or...
```json
{
  "code": 500,
  "declarationFullId": "uuid-xxxx",
  "messsage": "failed to save BIG-IP config (connection to http://admin:XXXXXX@localhost:8100/mgmt/tm/task/sys/config create task save sys config timed out)"
 }
```
Or another interesting one
```json
{
  "code": "ETIMEDOUT",
  "message": "failure querying config for tenant Common (connection to http://admin:XXXXXX@localhost:8100/mgmt/tm/asm/policies?$filter=partition%20eq%20Common getting BIG-IP config timed out)",
  "host": "localhost",
  "tenant": "Common"
}
```

### Expected Behavior
The declaration to deploy successfully

### Actual Behavior
The deployment times out.

### Notes
We have followed the best practices guidelines in terms of provisioning and all sorts of things. It's not doing any difference. The tenants we are deploying are small(!) (we are talking about a partition with a single irule for the last one that received "ETIMEDOUT" error).

**Comment by dstokesf5** [comment]
Unfortunately, there is little we can do if the underlying framework is becoming unresponsive. However, I have added AUTOTOOL-2201 to our internal backlog as a feature request to investigate and add retries to the 4  cases you mentioned.

**Comment by mdditt2000** [comment]
reopening and prioritized due to customer requiring urgent ix

**Comment by vidya-sagar-m** [comment]
Facing the timeout errors on the bulk deployment of AS3 Declarations. Below are the logs:
```
Mon, 12 Dec 2022 11:33:34 GMT - fine: [appsvcs] {"message":"audit E15","level":"debug"}
Mon, 12 Dec 2022 11:33:34 GMT - fine: [appsvcs] {"message":"Begin parsing desired E15 config","level":"debug"}
Mon, 12 Dec 2022 11:33:34 GMT - config: [appsvcs] {"message":"parsing desired E15 config time: 0.136 seconds","level":"notice"}
Mon, 12 Dec 2022 11:33:34 GMT - fine: [appsvcs] {"message":"Begin parsing current E15 config","level":"debug"}
Mon, 12 Dec 2022 11:33:34 GMT - fine: [appsvcs] {"message":"tenant E15 lacks a partition","level":"debug"}
Mon, 12 Dec 2022 11:33:34 GMT - config: [appsvcs] {"message":"parsing current E15 config time: 0.019 seconds","level":"notice"}
Mon, 12 Dec 2022 11:33:34 GMT - fine: [appsvcs] {"message":"Begin generating E15 diff","level":"debug"}
Mon, 12 Dec 2022 11:33:34 GMT - config: [appsvcs] {"message":"generating E15 diff time: 0.048 seconds","level":"notice"}
Mon, 12 Dec 2022 11:33:34 GMT - fine: [appsvcs] {"message":"Begin updating E15","level":"debug"}
Mon, 12 Dec 2022 11:33:34 GMT - fine: [appsvcs] {"message":"generating tmsh commands and pre-script iControlRest calls","level":"debug"}
Mon, 12 Dec 2022 11:33:34 GMT - fine: [appsvcs] {"message":"iControl command","data":["/E15/AP15_vip15_com_http_tcp_443/Cert_AP15_vip15_com.crt"],"level":"debug"}
Mon, 12 Dec 2022 11:33:34 GMT - fine: [appsvcs] {"message":"iControl command","data":["/E15/AP15_vip15_com_http_tcp_443/Cert_AP15_vip15_com.key"],"level":"debug"}
Mon, 12 Dec 2022 11:33:34 GMT - fine: [appsvcs] {"message":"getting whitelist","level":"debug"}
Mon, 12 Dec 2022 11:33:34 GMT - fine: [appsvcs] {"message":"setting whitelist","level":"debug"}
Mon, 12 Dec 2022 11:33:34 GMT - fine: [appsvcs] {"message":"executing pre-script iControl calls","level":"debug"}
Mon, 12 Dec 2022 11:33:34 GMT - fine: [appsvcs] {"message":"Executing pre-script time: 20 milliseconds","level":"debug"}
Mon, 12 Dec 2022 11:33:34 GMT - fine: [appsvcs] {"message":"uploading cli script","level":"debug"}
Mon, 12 Dec 2022 11:33:34 GMT - fine: [appsvcs] {"message":"Uploading time: 8 milliseconds","level":"debug"}
Mon, 12 Dec 2022 11:33:34 GMT - fine: [appsvcs] {"message":"merging cli script","level":"debug"}
Mon, 12 Dec 2022 11:33:35 GMT - fine: [appsvcs] {"message":"Merging time: 634 milliseconds","level":"debug"}
Mon, 12 Dec 2022 11:33:35 GMT - fine: [appsvcs] {"message":"running cli script","level":"debug"}
Mon, 12 Dec 2022 11:33:42 GMT - fine: [appsvcs] {"message":"Running time: 6927 milliseconds","level":"debug"}
Mon, 12 Dec 2022 11:33:42 GMT - fine: [appsvcs] {"message":"verifying cli script","level":"debug"}
Mon, 12 Dec 2022 11:33:42 GMT - fine: [appsvcs] {"message":"Verify time: 291 milliseconds","level":"debug"}
Mon, 12 Dec 2022 11:33:42 GMT - fine: [appsvcs] {"message":"overwriting finished cli script","level":"debug"}
Mon, 12 Dec 2022 11:33:42 GMT - fine: [appsvcs] {"message":"Overwriting time: 21 milliseconds","level":"debug"}
Mon, 12 Dec 2022 11:33:42 GMT - severe: [appsvcs] {"message":"Declaration failed: GET http://admin:XXXXXX@localhost:8100/mgmt/tm/task/cli/script/1670844815228324 checking cli script status response=500 body={\"code\":500,\"message\":\"Connection refused\",\"referer\":\"Unknown\",\"restOperationId\":4665275,\"kind\":\":resterrorresponse\"}","level":"error"}
Mon, 12 Dec 2022 11:33:42 GMT - config: [appsvcs] {"message":"updating E15 time: 8.114 seconds","level":"notice"}
Mon, 12 Dec 2022 11:33:42 GMT - fine: [appsvcs] {"code":422,"message":"declaration failed","response":"GET http://admin:XXXXXX@localhost:8100/mgmt/tm/task/cli/script/1670844815228324 checking cli script status response=500 body={\"code\":500,\"message\":\"Connection refused\",\"referer\":\"Unknown\",\"restOperationId\":4665275,\"kind\":\":resterrorresponse\"}","host":"localhost","tenant":"E15","runTime":8337,"level":"debug"}
Mon, 12 Dec 2022 11:33:42 GMT - fine: [appsvcs] {"message":"audit E16","level":"debug"}
Mon, 12 Dec 2022 11:33:42 GMT - fine: [appsvcs] {"message":"Begin parsing desired E16 config","level":"debug"}
Mon, 12 Dec 2022 11:33:42 GMT - config: [appsvcs] {"message":"parsing desired E16 config time: 0.068 seconds","level":"notice"}
Mon, 12 Dec 2022 11:33:42 GMT - fine: [appsvcs] {"message":"Begin parsing current E16 config","level":"debug"}
Mon, 12 Dec 2022 11:33:42 GMT - fine: [appsvcs] {"message":"tenant E16 lacks a partition","level":"debug"}
Mon, 12 Dec 2022 11:33:42 GMT - config: [appsvcs] {"message":"parsing current E16 config time: 0.196 seconds","level":"notice"}
Mon, 12 Dec 2022 11:33:42 GMT - fine: [appsvcs] {"message":"Begin generating E16 diff","level":"debug"}
Mon, 12 Dec 2022 11:33:42 GMT - config: [appsvcs] {"message":"generating E16 diff time: 0.034 seconds","level":"notice"}
Mon, 12 Dec 2022 11:33:42 GMT - fine: [appsvcs] {"message":"Begin updating E16","level":"debug"}
Mon, 12 Dec 2022 11:33:42 GMT - fine: [appsvcs] {"message":"generating tmsh commands and pre-script iControlRest calls","level":"debug"}
Mon, 12 Dec 2022 11:33:42 GMT - fine: [appsvcs] {"message":"iControl command","data":["/E16/AP16_vip16_com_http_tcp_443/Cert_AP16_vip16_com.crt"],"level":"debug"}
Mon, 12 Dec 2022 11:33:42 GMT - fine: [appsvcs] {"message":"iControl command","data":["/E16/AP16_vip16_com_http_tcp_443/Cert_AP16_vip16_com.key"],"level":"debug"}
Mon, 12 Dec 2022 11:33:42 GMT - fine: [appsvcs] {"message":"getting whitelist","level":"debug"}
Mon, 12 Dec 2022 11:33:42 GMT - fine: [appsvcs] {"message":"setting whitelist","level":"debug"}
Mon, 12 Dec 2022 11:33:42 GMT - fine: [appsvcs] {"message":"executing pre-script iControl calls","level":"debug"}
Mon, 12 Dec 2022 11:33:42 GMT - fine: [appsvcs] {"message":"Executing pre-script time: 27 milliseconds","level":"debug"}
Mon, 12 Dec 2022 11:33:42 GMT - fine: [appsvcs] {"message":"uploading cli script","level":"debug"}
Mon, 12 Dec 2022 11:33:42 GMT - fine: [appsvcs] {"message":"Uploading time: 12 milliseconds","level":"debug"}
Mon, 12 Dec 2022 11:33:42 GMT - fine: [appsvcs] {"message":"merging cli script","level":"debug"}
Mon, 12 Dec 2022 11:33:43 GMT - fine: [appsvcs] {"message":"Merging time: 610 milliseconds","level":"debug"}
Mon, 12 Dec 2022 11:33:43 GMT - fine: [appsvcs] {"message":"running cli script","level":"debug"}
Mon, 12 Dec 2022 11:33:48 GMT - fine: [appsvcs] {"message":"Running time: 4557 milliseconds","level":"debug"}
Mon, 12 Dec 2022 11:33:48 GMT - fine: [appsvcs] {"message":"verifying cli script","level":"debug"}
Mon, 12 Dec 2022 11:33:48 GMT - fine: [appsvcs] {"message":"Verify time: 257 milliseconds","level":"debug"}
Mon, 12 Dec 2022 11:33:48 GMT - fine: [appsvcs] {"message":"overwriting finished cli script","level":"debug"}
Mon, 12 Dec 2022 11:33:48 GMT - fine: [appsvcs] {"message":"Overwriting time: 30 milliseconds","level":"debug"}
Mon, 12 Dec 2022 11:33:48 GMT - config: [appsvcs] {"message":"updating E16 time: 5.576 seconds","level":"notice"}
Mon, 12 Dec 2022 11:33:48 GMT - fine: [appsvcs] {"code":200,"message":"success","lineCount":39,"host":"localhost","tenant":"E16","runTime":5891,"level":"debug"}
```

API response:
```json
{
   "code":422,
   "message":"declaration failed",
   "response":"GET http://admin:XXXXXX@localhost:8100/mgmt/tm/task/cli/script/1670844815228324 checking cli script status response=500 body={\"code\":500,\"message\":\"Connection refused\",\"referer\":\"Unknown\",\"restOperationId\":4665275,\"kind\":\":resterrorresponse\"}",
   "host":"localhost",
   "tenant":"E15",
   "runTime":8337
}
```

**Comment by sunitharonan** [comment]
This has been resolved in the 3.42.0 release of AS3.
